src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineWindowsLauncher.java

```diff
@@ -0,0 +1,269 @@
+        // doesn't work that well.
+        boolean successful = false;
+        PrintStream logger = listener.getLogger();
+        logInfo(computer, listener, "Launching instance: " + node.getNodeName());
+        try {
+            boolean isBootstrapped = bootstrap(computer, listener);
+            if (isBootstrapped) {
+                // connect fresh as ROOT
+                logInfo(computer, listener, "connect fresh as root");
+                cleanupConn = connectToSsh(computer, listener);
+                if (!authenticateSSH(node.windowsConfig, cleanupConn, listener)) {
+                    logWarning(computer, listener, "Authentication failed");
+                    return; // failed to connect
+                }
+            } else {
+                logWarning(computer, listener, "bootstrapresult failed");
+                return;
+            }
+            conn = cleanupConn;
+
+            SCPClient scp = conn.createSCPClient();
+            String tmpDir = "C:\\";
+            logInfo(computer, listener, "Copying slave.jar to: " + tmpDir);
+            scp.put(Jenkins.getInstance().getJnlpJars("slave.jar").readFully(), "slave.jar", tmpDir);
+
+            // Confirm Java is installed
+            if (!testCommand(computer, conn, "java -fullversion", logger, listener)) {
+                logWarning(computer, listener, "Java is not installed.");
+            }
+
+            //TODO: allow jvmopt configuration
+            String launchString = "java -jar C:\\slave.jar";
+
+            logInfo(computer, listener, "Launching Jenkins agent via plugin SSH: " + launchString);
+            final Session sess = conn.openSession();
+            sess.execCommand(launchString);
+            computer.setChannel(sess.getStdout(), sess.getStdin(), logger, new Channel.Listener() {
+                @Override
+                public void onClosed(Channel channel, IOException cause) {
+                    sess.close();
+                    conn.close();
+                }
+            });
+        } catch (Exception e) {
+            logException(computer, listener, "Error getting exception", e);
+        }
+    }
+
+    private boolean testCommand(ComputeEngineComputer computer, Connection conn, String checkCommand, PrintStream logger, TaskListener listener)
+            throws IOException, InterruptedException {
+        logInfo(computer, listener, "Verifying: " + checkCommand);
+        return conn.exec(checkCommand, logger) == 0;
+
+    }
+
+    private boolean authenticateSSH(WindowsConfiguration windowsConfig, Connection sshConnection,
+                                    TaskListener listener) throws Exception{
+        boolean isAuthenticated;
+        String windowsUsername = windowsConfig.getWindowsUsername();
+        if (windowsConfig.getPrivateKeyCredentialsId() != null) {
+            StandardUsernameCredentials cred = windowsConfig.getPrivateKeyCredentials();
```

Comment:
No need to make this a variable, you can embed the getPrivateKeyCredentials() below.